configs/textrecog/sar/sar_r31_parallel_decoder_toy_dataset.py

```diff
@@ -102,18 +102,17 @@
     test_mode=True)
 
 data = dict(
-    samples_per_gpu=16,
```

Comment:
if samples_per_gpu for train/val/test are same, then you can specify once at this line.

Reply:
1. Currently global `data.samples_per_gpu` just for `train`, but not `val` and `test`. 
2. Since most of text detect models (use `MultiScaleFlipAug` in test_pipeline) and text recognition models (use `MultiRotateAugOCR` in test_pipeline) don't support batch inference.
3. The batch test will be activated only when setting `test_dataloader=dict(samples_per_gpu=8)`. It will disable `aug_test` for text recognition.

Reply:
> will disable aug_test

why this is necessary?

Reply:
> Currently global data.samples_per_gpu just for train, but not val and test.

This story is too complex. An alternative way is to make samples_per_gpu a global option, and explicitly overwrite samples_per_gpu=1 in val/test. 

Special case random things feels wrong

Reply:
`samples_per_gpu` be global setting